cabal-install/src/Distribution/Client/ProjectPlanOutput.hs

```diff
@@ -175,6 +177,20 @@ encodePlanAsJson distDirLayout elaboratedInstallPlan elaboratedSharedConfig =
             ] ++
             bin_file (compSolverName comp)
      where
+      -- | Only add build-info file location if the Setup.hs CLI
+      -- is recent enough to be able to generate build info files.
+      -- Otherwise, do not add the expected file location.
+      --
+      -- Consumers of `plan.json` can use the absence of this file location
+      -- to indicate that the given component uses `build-type: Custom`
+      -- with an old lib:Cabal version.
+      buildInfoFileLocation ::Maybe J.Pair
+      buildInfoFileLocation
+        | elabSetupScriptCliVersion elab < mkVersion [3, 7, 0, 0]
+        = Nothing
+        | otherwise
+        = Just ("build-info" J..= J.String (buildInfoPref dist_dir))
```

Comment:
I feel like it'd be better to have this be null instead of just missing unless you chose to do it this way for a particular reason?

Reply:
Of course, mental typo!